src/graphql/utils/fetchWithAbort.ts

```diff
@@ -0,0 +1,27 @@
+import fetch from 'node-fetch'
+import AbortController from 'abort-controller'
+
+export const MAX_REQUEST_TIME = 1200
+
+async function fetchWithAbort(endpoint: string, headers: Object = {}) {
+  const controller = new AbortController()
+
+  // Abort the fetch request when it takes too long
+  const timeout = setTimeout(() => {
+    controller.abort()
+    console.warn('ABORTED', endpoint) // For logging in Sentry
+  }, MAX_REQUEST_TIME)
+
+  return await fetch(endpoint, { ...headers, signal: controller.signal })
+    .then(res => {
+      clearTimeout(timeout) // The data is on its way, so clear the timeout
+
+      if (res.status !== 200) {
+        return { status: res.status, message: '' }
+      }
+      return res.json()
+    })
+    .catch((e: Error) => ({ status: e.name === 'AbortError' ? 504 : 500, message: e.message }))
```

Comment:
Should we not be returning this to the consumer as a rejected promise? It would be hard to determine the shape of the returned value if this function resolves with differently shaped objects.

Reply:
If we don't catch the error here, every resolver would need to check for the same error and construct an error return. Also, we can't use `Promise.all` that scenario and would have to use `Promise.allSettled` instead

Reply:
Hmmm, yeah I understand. I think it would make sense to reject these promises and use `allSettled` on the higher level to deal with the errors.